### ISSUE TITLE
Build script - sync community theme config.xml file to /config/xml/themes/

### ISSUE THREAD

**Issue by getdatakick** [thirtybees/thirtybees#589]
Thirtybees release contains outdated `/config/xml/themes/default.xml` config file. We should ensure that this file is in sync with `/themes/community-theme-default/config.xml`.

For consistency, we should also copy `/themes/community-theme-default/config.xml` file to `/config/xml/themes/community-theme-default.xml`

### More info

When any theme is installed, its `Config.xml` file is copied to `/config/xml/themes/<theme-directory>.xml`. Information from this file is later used when the theme is enabled or disabled. (install / uninstall module, set up hooks and positions, ...)

For **community-theme-default** theme there's an exception - configuration file can be found in two locations:
- `/config/xml/themes/community-theme-default.xml`
- `/config/xml/themes/default.xml`

see this code:

```php
        if (file_exists(_PS_ROOT_DIR_.'/config/xml/themes/'.$theme->directory.'.xml')) {
            $configFile = _PS_ROOT_DIR_.'/config/xml/themes/'.$theme->directory.'.xml';
        } elseif ($theme->name == 'community-theme-default') {
            $configFile = _PS_ROOT_DIR_.'/config/xml/themes/default.xml';
        }
```

**Comment by Traumflug** [comment]
Alright, I've split this huge commit in more readable chunks and put them into branch *[themeupload](https://github.com/thirtybees/thirtybees/commits/themeupload)*.

- ad102e575f3147bbb02f6844ec015286403cb723 is already on its way to *1.0.x*.
- 7ca127ebab0a8ec5c7f3f801c435cfa26a773c7b appears to be fine as well and trusting comments in the big commit there's not much point in testing.
- 21955513be6f24741b47543c27ce22181a97f49b needs testing for the split not breaking things.
- 89964b6def8c0eba43d50ee71ca0e984ab1667f9 is the essence of the whole thing. Looks good. Code more straightforward, less duplication.
- c7e7a15c41acc0436f907dd462882580d7268dcb is a nitpicking.
- a9e4ce209b9a46911ccea31b527fcf8d31eb9c7d eliminates one of the just introduced methods for simplicity.
- 8841df7c18f16ac86999df15b127d357f93d9584 is the configuration file change. This undoubtly needs careful testing. Potential to change the entire shop appearance and chances are good for configuration in the theme to be much older than the one in core.

Does this split find your agreement, @getdatakick ? I made sure to keep the commit author.

**Comment by getdatakick** [comment]
sure, I have no problems with that. Indeed it's more clear this way what each commit is supposed to fix (or break :)

**Comment by Traumflug** [comment]
This was solved somewhere between 1.0.8 and 1.1.0. Theme configurations in _config/xml/themes/_ are gone in favor of the one in the theme its self. Retrocompatibility code moves existing ones back into the theme folder. Theme installation code was adjusted to find the right _config.xml_ in theme distribution packages and place that in the theme folder.

If there's anything left, please don't hesitate to reopen this issue or open another one.
